core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -78,28 +78,45 @@ private[kafka] abstract class SubSourceLogic[K, V, Msg](
   private implicit val askTimeout = Timeout(10000, TimeUnit.MILLISECONDS)
   def partitionAssignedCB(tps: Set[TopicPartition]) = {
     implicit val ec = materializer.executionContext
-    getOffsetsOnAssign.fold(pumpCB.invoke(tps)) { getOffsets =>
-      getOffsets(tps)
+
+    val topicsToBeAssigned = tps -- partitionsToRevoke
+    partitionsToRevoke = partitionsToRevoke -- tps
+
+    getOffsetsOnAssign.fold(pumpCB.invoke(topicsToBeAssigned)) { getOffsets =>
+      getOffsets(topicsToBeAssigned)
```

Comment:
Good idea to restrict to the "new" ones.
Wouldn't `topicPartitions` or `partitions` be a better name?

Reply:
Totally agree. Changed to `partitions`